src/lib.rs

```diff
@@ -75,17 +75,17 @@ impl Xor8 {
     /// Populate fills the filter with provided keys.
     ///
     /// The caller is responsible to ensure that there are no duplicate keys.
-    pub fn populate(keys: &Vec<u64>) -> Box<Xor8> {
+    pub fn new(keys: &Vec<u64>) -> Self {
```

Comment:
I feel populate() is more idiomatic. new() typically means empty instantiation.

Reply:
But as the [original article](https://lemire.me/blog/2019/12/19/xor-filters-faster-and-smaller-than-bloom-filters/) points out, there's no such thing as "empty Xor filter" — it is immutable only exists in its populated form!

In my mental picture, `new()` should mean empty instantiation only if there's some lifecycle involved after it happens. I have a feeling this is also how it works in Rust standard library: `Vec` has a potentially complicated lifecycle, items might get added or removed — this is why `Vec::new()` instantiates an empty vector. But, say, for `Box` not having a pointer inside makes little sense, and this is why `Box::new` takes an argument and returns a populated Box.
(What is even funnier, in `Box` case pointing to some unintialized memory _might_ make sense in some cases, and there's even a special, not yet stabilized, `Box::new_uninit` method which does that. Hopefully, this illustrates my point of `new` doing the most ergonomically-expected thing by default, not some semantically-uniformed one).

Does this reasoning make any sense to you?

Reply:
:) you _are_ convincing. My own reasoning went with plain english. Btw, by changing the public method signature aren't we going to break existing code ? Also, the origina-code in golang and other rust implementation is sticking to populate() method.

Reply:
According to [semver specification](https://semver.org/#spec-item-4) anything may break at any moment before we claim the first major release. We're also not on Crates yet, afaik — so we're not likely to break anything in practice either =).

As for the alternative implementation, I only found [this one](https://github.com/ayazhafiz/xorf/), which [implements `From`](https://github.com/ayazhafiz/xorf/blob/fe5f794d8a3934b4213e722cc69b92d7312a44d3/src/xor8.rs#L69-L79) for vectors and slices, so the filters can be created implicitly by calling `into()` in the appropriate place.
I'm not a fan of time-consuming `From` implementations — but this is my personal preference, and I admit that it's at least an elegant approach.
AFAIU Go implementation does what it does because golang is _anti-elegant_ by design. It is not even a bad trait in general — but I don't think that in API with two straightforward methods (and in a world where docs.rs exists) following the exact naming convention from Go does make too much sense.